packages/workspace/src/browser/workspace-variable-contribution.ts

```diff
@@ -34,6 +36,58 @@ export class WorkspaceVariableContribution implements VariableContribution {
                 return uri ? uri.displayName : undefined;
             }
         });
+        variables.registerVariable({
+            name: 'file',
+            description: 'The path of the currently opened file',
+            resolve: () => {
+                const uri = this.getCurrentURI();
+                return uri ? uri.path.toString() : undefined;
+            }
+        });
+        variables.registerVariable({
+            name: 'fileBasename',
+            description: 'The basename of the currently opened file',
+            resolve: () => {
+                const uri = this.getCurrentURI();
+                return uri ? uri.path.base : undefined;
+            }
+        });
+        variables.registerVariable({
+            name: 'fileBasenameNoExtension',
+            description: "The currently opened file's name without extension",
+            resolve: () => {
+                const uri = this.getCurrentURI();
+                return uri ? uri.path.name : undefined;
+            }
+        });
+        variables.registerVariable({
+            name: 'fileDirname',
+            description: "The name of the currently opened file's directory",
+            resolve: () => {
+                const uri = this.getCurrentURI();
+                return uri ? uri.path.dir.toString() : undefined;
+            }
+        });
+        variables.registerVariable({
+            name: 'fileExtname',
+            description: 'The extension of the currently opened file',
+            resolve: () => {
+                const uri = this.getCurrentURI();
+                return uri ? uri.path.ext : undefined;
+            }
+        });
+        variables.registerVariable({
+            name: 'relativeFile',
+            description: "The currently opened file's path relative to the workspace root",
+            resolve: async () => {
+                const currentURI = this.getCurrentURI();
+                if (!currentURI) {
+                    return undefined;
+                }
+                const workspaceRelativePath = this.getWorkspaceRelativePath(currentURI);
```

Comment:
`await` or just `return this.getWorkspaceRelativePath(currentURI)`, otherwise `workspaceRelativePath` is a promise

Reply:
Thanks. Fixed.